lib/sibyl.js

```diff
@@ -68,34 +67,164 @@ Sibyl.prototype.check = function (address, cb) {
   })
 }
 
-Sibyl.prototype._fetchFile = function (address, cb) {
+Sibyl.prototype.open = function (address, cb) {
+  this.emit('open')
+
+  assert.equal(typeof cb, 'function', 'sibyl.open: cb should be type function')
+
+  this._initialize(address, (err, protocol, location, version) => {
+    if (err) cb(err)
+    else {
+      this._fetch(protocol, location, version, (err) => {
+        if (err) cb(err)
+        else this._finalize(cb)
+      })
+    }
+  })
+}
+
+Sibyl.prototype._initialize = function (address, cb) {
+  var protocol = null
+  var location = null
+  var version = null
+
+  if (address) {
+    var match1 = address.match(/^([^@]+)?(@(.*))?/)
+    if (match1[1]) {
+      var match2 = match1[1].match(/^([\w]+):\/\/(.+)/)
+      protocol = match2[1] || null
+      location = match2[2] || null
+    }
+    version = match1[3] || null
+  }
+
+  this.directory = process.cwd()
+  var configDir = path.join(this.directory, '.sibyl')
+  fs.access(configDir, fs.constants.R_OK, (err) => {
+    if (err) {
+      if (location) {
+        // Create a new bundle as a child of the current directory
+        // SHA1 added to name to reduce chance of collisions caused by replacement of characters
```

Comment:
even if sha1 isn't used for cryptographic purposes here, we should probably still use sha256/sha512